tests/v2/actions/etcdsnapshot/creates.go

```diff
@@ -0,0 +1,709 @@
+package etcdsnapshot
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	apisV1 "github.com/rancher/rancher/pkg/apis/provisioning.cattle.io/v1"
+	rkev1 "github.com/rancher/rancher/pkg/apis/rke.cattle.io/v1"
+	"github.com/rancher/rancher/tests/v2/actions/scalinginput"
+	"github.com/rancher/rancher/tests/v2/actions/services"
+	deploy "github.com/rancher/rancher/tests/v2/actions/workloads/deployment"
+	"github.com/rancher/shepherd/clients/rancher"
+	management "github.com/rancher/shepherd/clients/rancher/generated/management/v3"
+	steveV1 "github.com/rancher/shepherd/clients/rancher/v1"
+	"github.com/rancher/shepherd/extensions/clusters"
+	"github.com/rancher/shepherd/extensions/clusters/kubernetesversions"
+	extdefault "github.com/rancher/shepherd/extensions/defaults"
+	"github.com/rancher/shepherd/extensions/defaults/stevetypes"
+	shepherdsnapshot "github.com/rancher/shepherd/extensions/etcdsnapshot"
+	extensionsingress "github.com/rancher/shepherd/extensions/ingresses"
+	nodestat "github.com/rancher/shepherd/extensions/nodes"
+	"github.com/rancher/shepherd/extensions/workloads"
+	"github.com/rancher/shepherd/extensions/workloads/pods"
+	namegen "github.com/rancher/shepherd/pkg/namegenerator"
+	"github.com/sirupsen/logrus"
+	v1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	networking "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	InitialIngress  = "ingress-before-restore"
+	InitialWorkload = "wload-before-restore"
+
+	all                          = "all"
+	containerImage               = "nginx"
+	containerName                = "nginx"
+	defaultNamespace             = "default"
+	DeploymentSteveType          = "apps.deployment"
+	isCattleLabeled              = true
+	ingressSteveType             = "networking.k8s.io.ingress"
+	ingressPath                  = "/index.html"
+	K3S                          = "k3s"
+	kubernetesVersion            = "kubernetesVersion"
+	namespace                    = "fleet-default"
+	port                         = "port"
+	postWorkload                 = "wload-after-backup"
+	ProvisioningSteveResouceType = "provisioning.cattle.io.cluster"
+	RKE1                         = "rke1"
+	RKE2                         = "rke2"
+	serviceAppendName            = "service-"
+	serviceType                  = "service"
+	windowsContainerImage        = "mcr.microsoft.com/windows/servercore/iis"
+	windowsContainerName         = "iis"
+)
+
+// CreateAndValidateSnapshotRestore is an e2e helper that determines the engine type of the cluster, then takes a snapshot, and finally restores the cluster
+// to the original snapshot. Resources are deployed at various stages to validate the snapshot process
+func CreateAndValidateSnapshotRestore(client *rancher.Client, clusterName string, etcdRestore *Config, containerImage string) error {
```

Comment:
ack. But I don't think refactoring is necessary, really. This is a helper function to do both creating and validation, which should still fit our rules for being an action. I'll wait for Hostbusters input for a second opinion

Reply:
Agreed refactoring shouldn’t happen in this PR to avoid the risk of breaking existing tests, with how involved the snapshot suite is. but, I think it would be beneficial to create a ticket to revisit this later. We can discuss the potential risks and benefits of separating the Create and Validate logic in our design call, to make sure the tests are easily maintainable.

Reply:
https://github.com/rancher/qa-tasks/issues/1582